modules/minimum-required-permissions-upi-azure.adoc

```diff
@@ -0,0 +1,241 @@
+====
+
+.Optional permissions for enabling user-managed encryption
+[%collapsible]
+====
+* `Microsoft.Compute/diskEncryptionSets/read`
+* `Microsoft.Compute/diskEncryptionSets/write`
+* `Microsoft.Compute/diskEncryptionSets/delete`
+* `Microsoft.KeyVault/vaults/read`
+* `Microsoft.KeyVault/vaults/write`
+* `Microsoft.KeyVault/vaults/delete`
+* `Microsoft.KeyVault/vaults/deploy/action`
+* `Microsoft.KeyVault/vaults/keys/read`
+* `Microsoft.KeyVault/vaults/keys/write`
+* `Microsoft.Features/providers/features/register/action`
+====
+
+The following permissions are required for deleting an {product-title} cluster on Microsoft Azure.
+
+.Required permissions for deleting authorization resources
+[%collapsible]
+====
+* `Microsoft.Authorization/roleAssignments/delete`
+====
+
+.Required permissions for deleting compute resources
+[%collapsible]
+====
+* `Microsoft.Compute/disks/delete`
+* `Microsoft.Compute/galleries/delete`
+* `Microsoft.Compute/galleries/images/delete`
+* `Microsoft.Compute/galleries/images/versions/delete`
+* `Microsoft.Compute/virtualMachines/delete`
+* `Microsoft.Compute/images/delete`
+====
+
+.Required permissions for deleting identity management resources
+[%collapsible]
+====
+* `Microsoft.ManagedIdentity/userAssignedIdentities/delete`
+====
+
+.Required permissions for deleting network resources
+[%collapsible]
+====
+* `Microsoft.Network/dnszones/read`
+* `Microsoft.Network/dnsZones/A/read`
+* `Microsoft.Network/dnsZones/A/delete`
+* `Microsoft.Network/dnsZones/CNAME/read`
+* `Microsoft.Network/dnsZones/CNAME/delete`
+* `Microsoft.Network/loadBalancers/delete`
+* `Microsoft.Network/networkInterfaces/delete`
+* `Microsoft.Network/networkSecurityGroups/delete`
+* `Microsoft.Network/privateDnsZones/read`
+* `Microsoft.Network/privateDnsZones/A/read`
+* `Microsoft.Network/privateDnsZones/delete`
+* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete`
+* `Microsoft.Network/publicIPAddresses/delete`
+* `Microsoft.Network/virtualNetworks/delete`
+====
+
+.Required permissions for checking the health of resources
+[%collapsible]
+====
+* `Microsoft.Resourcehealth/healthevent/Activated/action`
+* `Microsoft.Resourcehealth/healthevent/Resolved/action`
+* `Microsoft.Resourcehealth/healthevent/Updated/action`
+====
+
+.Required permissions for deleting resource group
+[%collapsible]
+====
+* `Microsoft.Resources/subscriptions/resourcegroups/delete`
+====
+
+.Required permissions for deleting storage resources
+[%collapsible]
+====
+* `Microsoft.Storage/storageAccounts/delete`
+* `Microsoft.Storage/storageAccounts/listKeys/action`
+====
+
+[NOTE]
+====
+You can create a custom role with all the permissions and assign that custom role to the service principal. To create a custom role in the Azure portal, see link:https://learn.microsoft.com/en-us/azure/role-based-access-control/custom-roles[Azure custom roles] in the Azure documentation.
+====
```

Comment:
I think this note is not needed since you are explaining it at the start of the module.

Reply:
Addressed